### PR TITLE
 utils/transcripts.py for chunking preserving previous behaviour as default for #8087

### LINKED ISSUE #8087: YoutubeLoader to load documents from transcripts as chunks
### Feature request

Currently, The YoutubeLoader class loads transcripts and converts all of them into a single large piece of text loosing all the information regarding when it was said in the video to be able to reference it later.
I understand we cannot have very small/ unpridictable lengths of transcripts, which is why it would be great to chunk them according to a input duration by user and retain some information regarding when its said. which get the best of both worlds by having large enough data and also retaining context

### Motivation

I was trying to make a vedio qa retriever, Then I noticed how the transcripts were being loaded, I have also noticed this issue with other transcript loaders such as VTT, then I had to make my custom function to load the transcripts in chunks of duration that made sense to me, which lead me to making this issue

### Your contribution

I already have the code for chunking the transcripts, would love to submit a PR.

### PR DESCRIPTION
**Pull request by beaterblank**
fixes #8087 
added chunking mechanism for for transcript loaders srt.py and youtubeloader.py so user could optionally load transcript for a certain duration

### CODE REVIEW

**Comment by beaterblank** [comment]
Hey @baskaryan  @eyurtsev and @rlancemartin why were some checks not sucessfull? any idea of what can I do?
Thank you!

**Comment by baskaryan** [comment]
> Hey @baskaryan @eyurtsev and @rlancemartin why were some checks not sucessfull? any idea of what can I do?
> Thank you!
Screenshot 2023-07-31 at 5 00 37 PM

looks like some linting errors. you can run `make lint` from the libs/langchain directory to see linting issues locally! and `make format` to automatically fix any formatting issues

File: libs/langchain/langchain/document_loaders/youtube.py [hunk 1]
```diff
@@ -177,7 +179,7 @@ def from_youtube_url(cls, youtube_url: str, **kwargs: Any) -> YoutubeLoader:
         video_id = cls.extract_video_id(youtube_url)
         return cls(video_id, **kwargs)
 
-    def load(self) -> List[Document]:
+    def load(self, duration: Optional[float] = None) -> List[Document]:
```
Comment (hwchase17) [inline libs/langchain/langchain/document_loaders/youtube.py:182]: this should not be passed into load

**Review by hwchase17 — CHANGES_REQUESTED** [review]
this should be an optional parameter and try to maintain backwards compat

File: libs/langchain/langchain/document_loaders/youtube.py (hunk 1 above)
Comment (beaterblank) [reply libs/langchain/langchain/document_loaders/youtube.py:182]: Hey It is an optinal variable and when its not given its defaulted to None and that results in previous behaiviour, that was my intention, I'm not sure how i would be able to achieve that without the duration variable

Comment (beaterblank) [reply libs/langchain/langchain/document_loaders/youtube.py:181]: okay so ill just move the duration to init method and reference it from there got it thank you!

**Comment by leo-gan** [comment]
@beaterblank Hi , could you, please, resolve the merging issues and address the last comments (if needed)? After that, ping me and I push this PR for the review. Thanks!

**Comment by beaterblank** [comment]
Hey sure yeah, thank you!

**Comment by beaterblank** [comment]
@leo-gan Hi, the merge issue was regarding updating the branch, and I have updated my branch, 

regarding previous comments I was told that 

 - all configuration information is passed via the initializer by convention 

so I addressed that in my previous commit removing duration from load function and I've put it in the initializer as per the convention

**Comment by leo-gan** [comment]
@eyurtsev Please, review it. Thanks

**Comment by beaterblank** [comment]
@eyurtsev please review, Thanks.

**Comment by w4rlock999** [comment]
any update on this please? @beaterblank

**Comment by beaterblank** [comment]
> any update on this please? @beaterblank this is so cool if successfully merged!

@w4rlock999, I'm waiting for it to be reviewed  and merged, I made the requested changes still waiting to be merged.

File: libs/langchain/langchain/document_loaders/srt.py [hunk 2]
```diff
@@ -17,11 +20,30 @@ def __init__(self, file_path: str):
             )
         self.file_path = file_path
 
-    def load(self) -> List[Document]:
+    def load(self, duration: Optional[int] = None) -> List[Document]:
```
Comment (efriis) [inline libs/langchain/langchain/document_loaders/srt.py:23]: same - should be an init param, not a load param

**Review by efriis — COMMENTED** [review]
And please resolve the lint errors :). You should be able to resolve them with `make format` in libs/langchain. You'll need ruff 0.1.5 installed (per the deps)

File: libs/langchain/langchain/document_loaders/srt.py (hunk 2 above)
Comment (beaterblank) [reply libs/langchain/langchain/document_loaders/srt.py:24]: Hello thank you for pointing out.

Comment (beaterblank) [reply libs/langchain/langchain/document_loaders/srt.py]: acb6c8cda2cf594ad592df4c316340218b2ffc33 resolves this

Comment (beaterblank) [reply libs/langchain/langchain/document_loaders/srt.py]: @efriis please review

**Pull request closed without merging**